Source/Particles/Gather/ScalarFieldGather.H

```diff
@@ -0,0 +1,112 @@
+/* Copyright 2021 Modern Electron
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef SCALARFIELDGATHER_H_
+#define SCALARFIELDGATHER_H_
+
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void compute_weights (const amrex::ParticleReal xp,
```

Comment:
These functions assume that the scalar field is defined at the cell nodes, as defined [here](https://amrex-codes.github.io/amrex/docs_html/Basics.html#id2). So if these are to be general routines, maybe the documentation and/or the function names should reflect this.

Reply:
Thanks for the suggestion. I have changed the function names to stress that it is for use with a nodal field and added more comments about that to the documentation.